petastorm/pytorch.py

```diff
@@ -58,7 +121,9 @@ def __iter__(self):
             # Default collate does not work nicely on namedtuples and treat them as lists
             # Using dict will result in the yielded structures being dicts as well
             row_as_dict = row._asdict()
-            batch.append(self.transform(row_as_dict) if self.transform else row_as_dict)
+            _sanitize_pytorch_types(row_as_dict)
+            transformed_row = self.transform(row_as_dict) if self.transform else row_as_dict
```

Comment:
The transforms should be called first, no? (otherwise the user doesnt have a chance to convert strings to something else before the `NotImplementedError` is raised in `_sanitize_pytorch_types`)

Reply:
Will not raise an error in case of strings/objects in `_sanitize_pytorch_types`. Will leave a chance for a user to update these types in transform. Added a test case that covers that scenario. Does it make sense?